### ISSUE TITLE
Not working on Android

### ISSUE THREAD

**Issue by tivewsalaeharad** [cdk/cdk#958]
Problems with class java.awt.Color, that cannot be resolved from Android environment. Does anyone have ideas, how to fix?

> E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.hand.space, PID: 11135
    java.lang.NoClassDefFoundError: Failed resolution of: [Ljava/awt/Color;
        at org.openscience.cdk.depict.DepictionGenerator.<clinit>(DepictionGenerator.java:121)
...

**Comment by egonw** [comment]
We had a similar problem a long time ago with Bioclipse which was based on SWT. At the time we used a generalization with AWT and SWT implementations. I need to dig up how that worked and why it no longer.

If you happen to use Gradle, maybe this is an alternative: https://imperceptiblethoughts.com/shadow/getting-started/ ?

**Comment by tivewsalaeharad** [comment]
@egonw I tried, but unfortunately build failed

> A problem occurred configuring project ':app'.
> Could not resolve all artifacts for configuration ':app:classpath'.
   > Could not find gradle.plugin.com.github.jengelman.gradle.plugins:shadow:7.1.2.
     Searched in the following locations:
       - https://plugins.gradle.org/m2/gradle/plugin/com/github/jengelman/gradle/plugins/shadow/7.1.2/shadow-7.1.2.pom
       - https://plugins.gradle.org/m2/gradle/plugin/com/github/jengelman/gradle/plugins/shadow/7.1.2/shadow-7.1.2.jar
     Required by:
         project :app
> Possible solution:
> - Declare repository providing the artifact, see the documentation at https://docs.gradle.org/current/userguide/declaring_repositories.html

**Comment by johnmay** [comment]
If you want to provide a patch feel free but I believe most stuff should work if you exclude the rendering libs.

**Comment by johnmay** [comment]
Didn't mean to close right away but am marking this for closure without further input. CDK is a Java Library if it doesn't work on Android, unfortunately that's that. If there is something simple we can do then it may be possible but I believe AWT is quite embedded in the rendering.

**Comment by egonw** [comment]
> but I believe most stuff should work if you exclude the rendering libs.

@tivewsalaeharad, this is worth exploring. Do you intend to use the graphics aspects?

**Comment by tivewsalaeharad** [comment]
Finally, decided to do in such way: I have downloaded all files from org.openscience.cdk library to my project and required files from java.awt.*, java.awt.geom.* and sun.awt.geom.* packages (and of course, corrected `import` statements). I need only SVG, so I've commented some parts of code. The most serious problem was with fonts, so I've created fake `MyGlyphVector` class, that generates `GeneralPath` of each letter in hardcode style.

Not, canonically, but it works

To creators: thank U for cdk library, it draws chemical structure excellently!
